### ISSUE TITLE
EXC_BAD_ACCESS Crash using 1.13.2

### ISSUE THREAD

**Issue by posix88** [ScreebApp/sdk-ios-public#3]
👋 Hi

We are experiencing crashes with the latest version of Screeb **1.13.2**

This is what Firebase is catching

EXC_BAD_ACCESS

```
Crashed: com.apple.main-thread
0  libobjc.A.dylib                0x3a7c objc_retain + 16
1  libobjc.A.dylib                0x3a7c objc_retain_x0 + 16
2  libswiftCore.dylib             0x402e9c objcRetainAndReturn(void*) + 32
3  Screeb                         0x1b504 _swift_stdlib_malloc_size + 2168
4  libswiftCore.dylib             0x399c48 initializeWithCopy for ClosedRange<>.Index + 196
5  Screeb                         0xf224 UIEdgeInsetsInsetRect + 14952
6  Screeb                         0xbf10 UIEdgeInsetsInsetRect + 1876
7  Screeb                         0x1a9b8 block_destroy_helper + 4292
8  Screeb                         0x158b44 block_destroy_helper + 8304
9  Screeb                         0x165688 block_destroy_helper + 288
10 Screeb                         0x172b88 __swift_allocate_boxed_opaque_existential_1 + 152
11 libdispatch.dylib              0x2320 _dispatch_call_block_and_release + 32
12 libdispatch.dylib              0x3eac _dispatch_client_callout + 20
13 libdispatch.dylib              0x126a4 _dispatch_main_queue_drain + 928
14 libdispatch.dylib              0x122f4 _dispatch_main_queue_callback_4CF + 44
15 CoreFoundation                 0x98c28 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
16 CoreFoundation                 0x7a560 __CFRunLoopRun + 1992
17 CoreFoundation                 0x7f3ec CFRunLoopRunSpecific + 612
18 GraphicsServices               0x135c GSEventRunModal + 164
19 UIKitCore                      0x39d6e8 -[UIApplication _run] + 888
20 UIKitCore                      0x39d34c UIApplicationMain + 340
```

Let me know if you need more info, keep in mind that we are experiencing this issue after upgrading to **1.13.2**

**Comment by samber** [comment]
Hi @posix88 

And thanks for the report.

I transmit it to our iOS developer and let will you know.

**Comment by emiliopavia** [comment]
It might be related also to this

<img width="511" alt="Screenshot 2023-06-29 alle 09 07 04" src="https://github.com/ScreebApp/sdk-ios-public/assets/467098/8acd6c3d-2f6b-4992-affd-1bd8c35089fd">

These were the latest logs before crashing

```
2023-06-28 19:06:04.641651+0200 App[41190:1699628] Screeb: Command Command CheckTarget has been called
2023-06-28 19:21:56.541212+0200 App[41190:1699628] Screeb: Command Command RefreshTarget has been called
2023-06-28 19:21:56.548447+0200 App[41190:1652529] Screeb: Request ⚡️ Request Started Optional(GET https://rpc.screeb.app/rpc/1.0/channel/30eb4a32-76b8-4187-af80-1f578050263c)
2023-06-28 19:21:56.548616+0200 App[41190:1652529] Screeb: Request ⚡️ Body Data Optional("None")
Object 0x600001e0daa0 deallocated with retain count 2, reference may have escaped from deinit.
2023-06-28 19:21:56.890881+0200 App[41190:1699628] Object 0x600001e0daa0 deallocated with retain count 2, reference may have escaped from deinit.
```

**Comment by Alexays** [comment]
Hi @emiliopavia and @posix88.
We have just released a version 2.0.0-rc.3 where the SDK has been rewritten from scratch to provide greater stability, instant updates of Screeb features and be even lighter.
Aswell all externals dependencies have been removed and the minimum iOS version has been reduced from 12.4 to 11.0.

Your crash should be fixed with this version, i invite you to test it :)

**Comment by posix88** [comment]
Thanks @Alexays I'll update the SDK and let you know

**Comment by cley44** [comment]
Hi @posix88,

We just released version 2.0.0 of our SDK, that should fix the issue.
It also includes a new feature called [hooks](https://github.com/ScreebApp/developers/wiki/iOS-SDK-Hooks) :)
